airbyte-webapp/src/components/connectorBuilder/Builder/InputsView.tsx

```diff
@@ -0,0 +1,312 @@
+  key: string;
+  definition: JSONSchema7;
+  required: boolean;
+  isNew?: boolean;
+  showDefaultValueField: boolean;
+  type: typeof supportedTypes[number];
+}
+
+function sluggify(str: string) {
+  return str.toLowerCase().replaceAll(/[^a-zA-Z\d]/g, "_");
+}
+
+function newInputInEditing(): InputInEditing {
+  return {
+    key: "",
+    definition: {},
+    required: false,
+    isNew: true,
+    showDefaultValueField: false,
+    type: "string",
+  };
+}
+
+function formInputToInputInEditing({ key, definition, required }: BuilderFormInput): InputInEditing {
+  const supportedType = supportedTypes.find((type) => type === definition.type) || "unknown";
+  return {
+    key,
+    definition,
+    required,
+    isNew: false,
+    showDefaultValueField: Boolean(definition.default),
+    type: supportedType !== "unknown" && definition.enum ? "enum" : supportedType,
+  };
+}
+
+function inputInEditingToFormInput({
+  type,
+  showDefaultValueField,
+  isNew,
+  ...values
+}: InputInEditing): BuilderFormInput {
+  return {
+    ...values,
+    definition: {
+      ...values.definition,
+      type: type === "enum" ? "string" : type === "unknown" ? values.definition.type : type,
+      // only respect the enum values if the user explicitly selected enum as type
+      enum: type === "enum" && values.definition.enum?.length ? values.definition.enum : undefined,
+      default: showDefaultValueField ? values.definition.default : undefined,
+    },
+  };
+}
+
+export const InputsView: React.FC = () => {
+  const { formatMessage } = useIntl();
+  const [inputs, , helpers] = useField<BuilderFormInput[]>("inputs");
+  const [inputInEditing, setInputInEditing] = useState<InputInEditing | undefined>(undefined);
+  const usedKeys = useMemo(() => inputs.value.map((input) => input.key), [inputs.value]);
+  const inputInEditValidation = useMemo(
+    () =>
+      yup.object().shape({
+        // make sure key can only occur once
+        key: yup
+          .string()
+          .notOneOf(
+            inputInEditing?.isNew ? usedKeys : usedKeys.filter((key) => key !== inputInEditing?.key),
+            "connectorBuilder.duplicateFieldID"
+          ),
+        required: yup.bool(),
+        definition: yup.object().shape({
+          title: yup.string().required("form.empty.error"),
+        }),
+      }),
+    [inputInEditing?.isNew, inputInEditing?.key, usedKeys]
+  );
+  return (
+    <BuilderConfigView heading={formatMessage({ id: "connectorBuilder.inputsTitle" })}>
+      <Text centered className={styles.inputsDescription}>
+        <FormattedMessage id="connectorBuilder.inputsDescription" />
+      </Text>
+      {inputs.value.length > 0 && (
+        <Card withPadding className={styles.inputsCard}>
+          <ol className={styles.list}>
+            {inputs.value.map((input) => (
+              <li className={styles.listItem} key={input.key}>
+                <Label className={styles.itemLabel}>{input.definition.title || input.key}</Label>
```

Comment:
One small issue with using a <Label> here that I forgot about when I suggested it: Label has a small bottom padding in its styles, which means the label is now not lined up with the edit button in the UI here:
<img width="679" alt="Screen Shot 2022-12-21 at 12 04 38 PM" src="https://user-images.githubusercontent.com/22731524/208993497-d448c44b-5683-43d6-b223-6d737e70e815.png">
maybe you can override that bottom padding here or refactor those styles to remove it?